addons/sale_loyalty/tests/test_program_with_code_operations.py

```diff
@@ -283,6 +283,68 @@ def test_on_next_order_reward_promotion_program_with_requirements(self):
         order_bis._update_programs_and_rewards()
         self.assertEqual(len(order_bis.order_line), 2, "Free product from a coupon generated from a promotion program on next order should not dissapear")
 
+    def test_partner_assigned_to_next_order_coupon(self):
+        # Step 1: Create a loyalty program of type 'next_order_coupons'
+        loyalty_program = self.env['loyalty.program'].create({
+            'name': '10% Discount on Next Order',
+            'program_type': 'next_order_coupons',
+            'applies_on': 'future',
+            'trigger': 'auto',
+            'rule_ids': [(0, 0, {})],
+            'reward_ids': [(0, 0, {
+                'reward_type': 'discount',
+                'discount': 10,
+                'discount_mode': 'percent',
+                'discount_applicability': 'order',
+            })],
+        })
+
+        # Step 2: Create a sale order
+        order = self.empty_order.copy()
+        self.product_A = self.env['product.product'].create({
+            'name': 'Product A',
+            'list_price': 100,
+            'sale_ok': True
+        })
+
+        order.write({'order_line': [
+            (0, False, {
+                'product_id': self.product_A.id,
+                'name': '1 Product A',
+                'product_uom': self.uom_unit.id,
+                'product_uom_qty': 1.0,
+            })
+        ]})
+
+        # Step 3: Apply the loyalty program to the order
+        result = order._try_apply_program(loyalty_program)
+
+        # Step 4: Check if the partner is assigned to the generated coupon
+        generated_coupons = result.get('coupon')
+        self.assertTrue(generated_coupons, "A coupon should have been generated")
+        for coupon in generated_coupons:
+            # Ensure the partner assigned to the coupon matches the order's partner
+            self.assertEqual(coupon.partner_id, order.partner_id, "The partner should be assigned to the 'next order coupon'")
+
+            # Modify or add additional fields
+            coupon.write({
+                "points": 1.0,
+                "point_name": "Coupon point(s)",
+                "points_display": "1 Coupon point(s)",
+            })
+
+        # Step 5: Verify that the generated coupon is in showable_rewards
+        # `_get_claimable_and_showable_rewards` is used in the website_sale_loyalty to show rewards during checkout
+        showable_rewards = order._get_claimable_and_showable_rewards()
+
+        coupon_found = False
+        for coupon, rewards in showable_rewards.items():
+            if coupon.id in [c.id for c in generated_coupons]:
+                coupon_found = True
+                break
```

Comment:
Isn't there any simpler way to do this like check coupon.id in key of `showable_rewords` :thinking: 

Also this test case should be in website_sale_loyalty since `_get_claimable_and_showable_rewards` is define in that module :)

Reply:
As per our discussion, if we are not changing the flow in the website sale, we can safely remove the check on whether the coupons are showing or not because there is no change in this PR that affects that flow.